### PR TITLE
Bratseth/segment yql by default take 2

### PR DESCRIPTION
**Pull request by bratseth**
@arnej27959 please review

Commit 1 you have already reviewed.
Commit 2 and 3 fixes the two bugs that surfaced.

### CODE REVIEW

File: container-search/src/main/java/com/yahoo/search/yql/YqlParser.java [hunk 1]
```diff
@@ -1329,8 +1331,8 @@ private Item instantiateWordItem(String field,
     }
 
     @SuppressWarnings({"deprecation"})
-    private boolean shouldSegment(String field, boolean fromQuery) {
-        return fromQuery && ! indexFactsSession.getIndex(field).isAttribute();
+    private boolean shouldSegment(String field, OperatorNode<ExpressionOperator> ast, boolean fromQuery) {
```
Comment (arnej27959) [inline container-search/src/main/java/com/yahoo/search/yql/YqlParser.java]: why do you take the "ast" as a parameter here?

**Review by arnej27959 — APPROVED** [review]
just one minor comment

File: container-search/src/main/java/com/yahoo/search/yql/YqlParser.java (hunk 1 above)
Comment (bratseth) [reply container-search/src/main/java/com/yahoo/search/yql/YqlParser.java]: To see if you are paying attention ... duly impressed.

Also, I thought I could use the getIndex(ast) method, which seems to do the right thing, but that makes further, not necessarily true assumptions. When this is safely out, I'm going to have another pass at this code to make it more understandable, so I didn't bother to clean up that parameter now.

**Comment by bratseth** [comment]
Also, let's not merge right now.

**Pull request merged**